### PR TITLE
Better Lying Down System (From White Dream)

### PR DESCRIPTION
**Pull request by VMSolidus**
# Description

Port of https://github.com/WWhiteDreamProject/wwdpublic/pull/2
And now also https://github.com/WWhiteDreamProject/wwdpublic/pull/8
Because Lying Down System is dependent on the Telescope System.

# TODO

- [x] Reconcile the code with core code, do code cleanup. I'll undraft this when I'm done. Probably not going to be tonight, because I will have to get some sleep soon to get up early for my calculus classes.

# Changelog

:cl: Spatison (White Dream)
- add: Added lying down system / Добавлена система лежания
- tweak: Lying down now uses do-afters that are visible to other people to indicate what is going on.
- add: Added telescope system / Добавлена система прицеливания
- tweak: Now you can aim from Hristov / Теперь можно прицеливаться из Христова

### CODE REVIEW

File: Content.Client/_White/Standing/LayingDownSystem.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        SubscribeLocalEvent<LayingDownComponent, MoveEvent>(OnMovementInput);
+
+        SubscribeNetworkEvent<CheckAutoGetUpEvent>(OnCheckAutoGetUp);
+    }
+
+    private void OnMovementInput(EntityUid uid, LayingDownComponent component, MoveEvent args)
+    {
+        if (!_timing.IsFirstTimePredicted
+            || !_standing.IsDown(uid)
+            || _buckle.IsBuckled(uid)
```
Comment (Mnemotechnician) [inline Content.Client/_White/Standing/LayingDownSystem.cs:31]: this should probably check ActionBlockerSystem.CanMove instead

**Comment by VMSolidus** [comment]
This is ready to go.

**Comment by ShatteredSwords** [comment]
Have you checked if this affects the traits like lethargy, parkour training, etc and the way they work?

**Comment by VMSolidus** [comment]
Apparently this isn't ready because it's now dependent on https://github.com/WWhiteDreamProject/wwdpublic/pull/8

**Comment by VMSolidus** [comment]
> Have you checked if this affects the traits like lethargy, parkour training, etc and the way they work?

image

Yes, the code used by these traits is still in use by the newly updated LayingDownSystem.

**Review by OldDanceJacket — APPROVED** [review]

**Review by FoxxoTrystan — COMMENTED** [review]
I just run a dev eviroment, press R...
Gliches happened.

After laying down and doing up when walking my character is back on the laying animation.

Intercations System may need to be looked as i forced someone down but the game still think their up.

image
(By the way in the screenshot im down even tho im up, normal walkspeed and all.)

I checkout, Build and test it 2 times, same results, so il not approve or request changes as well il be gone soon but im leaving a comment here.

**Comment by VMSolidus** [comment]
> I just run a dev eviroment, press R... Gliches happened.
> 
> After laying down and doing up when walking my character is back on the laying animation.
> 
> Intercations System may need to be looked as i forced someone down but the game still think their up.
> 
> image (By the way in the screenshot im down even tho im up, normal walkspeed and all.)
> 
> I checkout, Build and test it 2 times, same results, so il not approve or request changes as well il be gone soon but im leaving a comment here.

Thanks a bunch for testing it(I am usually severely behind in testing things). I'll take a look when I get a chance and see about getting the interaction system to play nice with the lying down system

File: Content.Shared/Input/ContentKeyFunctions.cs [hunk 2]
```diff
@@ -57,6 +57,7 @@ public static class ContentKeyFunctions
         public static readonly BoundKeyFunction ResetZoom = "ResetZoom";
         public static readonly BoundKeyFunction OfferItem = "OfferItem";
         public static readonly BoundKeyFunction ToggleStanding = "ToggleStanding";
+        public static readonly BoundKeyFunction LookUp = "LookUp"; // WD EDIT
```
Comment (DEATHB4DEFEAT) [inline Content.Shared/Input/ContentKeyFunctions.cs:60]:
```suggestion
        public static readonly BoundKeyFunction LookUp = "LookUp";
```

File: Resources/Locale/en-US/escape-menu/ui/options-menu.ftl [hunk 3]
```diff
@@ -269,3 +269,8 @@ ui-options-net-pvs-leave-tooltip = This limits the rate at which the client will
 ## Toggle window console command
 cmd-options-desc = Opens options menu, optionally with a specific tab selected.
 cmd-options-help = Usage: options [tab]
+
+## Combat Options
+ui-options-function-look-up = Look up/Take aim
+ui-options-function-auto-get-up = Automatically get up after falling
+ui-options-function-hold-look-up = Hold down the key to aim
```
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/escape-menu/ui/options-menu.ftl:276]:
```suggestion
ui-options-function-hold-look-up = Hold down the key to aim

```

File: Resources/Locale/ru-RU/Escape-Menu/options-menu.ftl [hunk 4]
```diff
@@ -0,0 +1,3 @@
+ui-options-function-look-up = Присмотреться/Прицелиться
+ui-options-function-auto-get-up = Автоматически вставать при падении
+ui-options-function-hold-look-up = Удерживать клавишу для прицеливания
```
Comment (DEATHB4DEFEAT) [inline Resources/Locale/ru-RU/Escape-Menu/options-menu.ftl:3]:
```suggestion
ui-options-function-hold-look-up = Удерживать клавишу для прицеливания

```

**Review by DEATHB4DEFEAT — APPROVED** [review]
Why is telescope a part of this?

**Pull request merged**